src/google/protobuf/compiler/js/js_generator.cc

```diff
@@ -2013,6 +2013,7 @@ void Generator::GenerateClass(const GeneratorOptions& options,
     printer->Print("\n");
     GenerateClassFieldInfo(options, printer, desc);
 
+    GenerateClassFromObject(options, printer, desc);
```

Comment:
We also need to turn on the `GENERATE_FROM_OBJECT` flag ?

Reply:
no; it defaults to `!goog.DISALLOW_TEST_ONLY_CODE` which we don't set AFAIK.
Tested downstream and `fromObject` works well!